src/utils/daos_hdlr.c

```diff
@@ -2425,7 +2425,7 @@ fs_copy(struct file_dfs *src_file_dfs,
 
 			D_ALLOC(buf, buf_size * sizeof(char));
 			if (buf == NULL)
-				return ENOMEM;
+				D_GOTO(out, rc = -DER_NOMEM);
```

Comment:
@daltonbohning I don't think the dst_filename and filename are cleaned up here if it returns in this case?

Reply:
@dsikich `dst_filename` and `filename` are free'ed at the bottom of the function. So as long as we use D_GOTO for error handling, we should be okay here.